padertorch/train/hooks.py

```diff
@@ -812,105 +816,115 @@ class StopTraining(Exception):
     pass
 
 
-class LossWeightAnnealingHook(TriggeredHook):
-    """
-    Anneals a loss weight within the los_weights dict of the trainer.
-    """
-    def __init__(self, name, factor, trigger, max_value=None, min_value=None):
+class AnnealingHook(TriggeredHook):
+    def __init__(self, trigger, breakpoints, name):
         """
+        Base class for piece-wise linear annealing. The piece-wise linear
+        function is parameterized by its breakpoints. Can also be used for
+        arbitrary annealing functions by stating breakpoints with an interval
+        similar to the trigger interval.
```

Comment:
Can you add a comment, what happens before the first breakpoint and what after the last breakpoint?

Reply:
done